.drone.yml

```diff
@@ -48,8 +48,144 @@ services:
 volumes:
   - name: dockersock
     temp: {}
+
+---
+kind: pipeline
+type: kubernetes
+name: publish
+
+trigger:
+  event:
+  - tag
+
+steps:
+  - name: fetch tags
+    image: docker:git
+    commands:
+      - git fetch --tags
+  - name: wait for docker
+    image: docker
+    commands:
+      - timeout 15s /bin/sh -c 'while [ ! -S /var/run/docker.sock ]; do sleep 1; done'
+      - docker version
+    volumes:
+      - name: dockersock
+        path: /var/run
+  - name: build buildbox
+    image: docker:git
+    commands:
+      - apk add --no-cache make
+      - make buildbox
+    volumes:
+      - name: dockersock
+        path: /var/run
+  - name: build tarball
+    image: docker:git
+    commands:
+      - apk add --no-cache make fakeroot
+      - make production
+    volumes:
+      - name: dockersock
+        path: /var/run
+  - name: publish
+    image: docker:git
+    environment:
+      REGISTRY_USERNAME:
+        from_secret: quay_username
+      REGISTRY_PASSWORD:
+        from_secret: quay_password
+      AWS_ACCESS_KEY_ID:
+        from_secret: AWS_ACCESS_KEY_ID
+      AWS_SECRET_ACCESS_KEY:
+        from_secret: AWS_SECRET_ACCESS_KEY
+      AWS_DEFAULT_REGION: us-east-1
+    commands:
+      - apk add --no-cache make aws-cli
```

Comment:
I'm assuming apk commands aren't cached (and isn't just in this particular pipeline). Just a weird question, have you considered setting up our own mirror or building our own containers with all deps included? Just one of those things where we're block users pulling our files constantly, and might be hitting upstream more than we really need to?

Anyways, not important for this PR, I'm just curious if you're thinking about anything here, or maybe drone is already able to cache this which would be cool.

Reply:
> have you considered setting up our own mirror or building our own containers with all deps included?

Yes.  Particularly for the docker:git + make combo that is used in just about every step.

Upsides:

* Faster runtime (make is ~1s per step, so maybe 4-5s per pipeline). aws-cli is much heavier,  maybe 10-15s per download.
* Image caching across steps
* Reduce reliance on 3rd party infrastructure in our product builds, we'd depend on bits we host instead.


Downsides:
* We now have a 2nd code base feeding into these builds that needs to be maintained.  (See docker-debian) -- I've invested 1-2 weeks in maintaining/migrating that code over the past year.
* A new cron to fetch & publish these on a schedule. This is a semi-hidden moving part that would likely cause delayed failures if it tips over.
* Monitoring/alerting for the above if we want to be proactive about them working.

In short, I don't see the ROI at the moment.  This is work that definitely has value, but also increases the scope of the Jenkins migration if I choose to implement it as part fo my current work.  For now, I've filed https://github.com/gravitational/ops/issues/185 to track the request.